nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/QueryDatabaseTable.java

```diff
@@ -0,0 +1,607 @@
+                if (!StringUtils.isEmpty(maxValue)) {
+                    Integer type = columnTypeMap.get(colName.toLowerCase());
+                    if (type == null) {
+                        // This shouldn't happen as we are populating columnTypeMap when the processor is scheduled.
+                        throw new IllegalArgumentException("No column type found for: " + colName);
+                    }
+                    // Add a condition for the WHERE clause
+                    whereClauses.add(colName + " > " + getLiteralByType(type, maxValue, preProcessStrategy));
+                }
+            }
+            if (!whereClauses.isEmpty()) {
+                query.append(" WHERE ");
+                query.append(StringUtils.join(whereClauses, " AND "));
+            }
+        }
+
+        return query.toString();
+    }
+
+    /**
+     * Returns a basic SELECT ... FROM clause with the given column names and table name. If no column names are found,
+     * the wildcard (*) is used to select all columns.
+     *
+     * @param tableName   The name of the table to select from
+     * @param columnNames A comma-separated list of column names to select from the table
+     * @return A SQL select statement representing a query of the given column names from the given table
+     */
+    protected StringBuilder getSelectFromClause(String tableName, String columnNames) {
+        final StringBuilder query = new StringBuilder("SELECT ");
+        if (StringUtils.isEmpty(columnNames) || columnNames.trim().equals("*")) {
+            query.append("*");
+        } else {
+            query.append(columnNames);
+        }
+        query.append(" FROM ");
+        query.append(tableName);
+        return query;
+    }
+
+    /**
+     * Returns a SQL literal for the given value based on its type. For example, values of character type need to be enclosed
+     * in single quotes, whereas values of numeric type should not be.
+     *
+     * @param type  The JDBC type for the desired literal
+     * @param value The value to be converted to a SQL literal
+     * @return A String representing the given value as a literal of the given type
+     */
+    protected String getLiteralByType(int type, String value, String preProcessStrategy) {
+        // Format value based on column type. For example, strings and timestamps need to be quoted
+        switch (type) {
+            // For string-represented values, put in single quotes
+            case CHAR:
+            case LONGNVARCHAR:
+            case LONGVARCHAR:
+            case NCHAR:
+            case NVARCHAR:
+            case VARCHAR:
+            case ROWID:
+            case DATE:
+            case TIME:
+                return "'" + value + "'";
+            case TIMESTAMP:
+                // Timestamp literals in Oracle need to be cast with TO_DATE
+                if ("Oracle".equals(preProcessStrategy)) {
```

Comment:
Would pull out the "Oracle" and create a static final member variable & reference that within the PropertyDescriptor as well